sw/device/lib/base/freestanding/limits.h

```diff
@@ -0,0 +1,51 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef SW_DEVICE_LIB_BASE_FREESTANDING_LIMITS_H_
+#define SW_DEVICE_LIB_BASE_FREESTANDING_LIMITS_H_
+
+#include <stdint.h>
+
+/**
+ * This header implements the limits.h standard header, as required by C11 S4p6.
+ * This header is specified in detail in S7.10 and S5.2.4.2.1 of the same.
+ */
+
+#define CHAR_BIT (8)
+#define SCHAR_MIN INT8_MIN
+#define SCHAR_MAX INT8_MAX
+#define UCHAR_MAX UINT8_MAX
+
+// NOTE: S5.2.4.2.1p2 specifies that the following defines should correspond to
+// the signedness of |char|; the RISCV ABI defines |char| to be unsigned. See
+// https://github.com/riscv/riscv-elf-psabi-doc/blob/master/riscv-elf.md#c-type-representation
+#define CHAR_MIN (0)
+#define CHAR_MAX UCHAR_MAX
+#define MB_LEN_MAX (1)
+
+// NOTE: the RISCV psABI defines the following widths for the integer types:
+// - short: 2
+// - int: 4
+// - long: <machine word>
```

Comment:
Can we update this comment to not abstract over machine word size? Our freestanding headers are only for RV32, and we are only using the ILP32 ABI, where long is defined to have size 4.